### PR TITLE
Bluetooth: Mesh: Remove extra rescheduling and ADV_FLAG_SCHEDULED flag in the extended advertiser

### PR DESCRIPTION
**Pull request by PavelVPV**
This PR removes extra rescheduling in the extended advertiser after the last advertising has been finished. This commit also removes the `ADV_FLAG_SCHEDULED` flag as not needed anymore.

This extra rescheduling and the flag were needed to delay the next advertisement by 20ms. Now, when the delay is removed the advertiser can pick the next advertisement from the pool and push to the host right away.

### CODE REVIEW

**Review by Andrewpini — DISMISSED** [review]

**Review by akredalen — DISMISSED** [review]

File: subsys/bluetooth/mesh/adv_ext.c [hunk 1]
```diff
@@ -325,8 +317,6 @@ static bool schedule_send(struct bt_mesh_ext_adv *ext_adv)
 	if (atomic_test_bit(ext_adv->flags, ADV_FLAG_ACTIVE)) {
 		atomic_set_bit(ext_adv->flags, ADV_FLAG_SCHEDULE_PENDING);
 		return false;
-	} else if (atomic_test_and_set_bit(ext_adv->flags, ADV_FLAG_SCHEDULED)) {
-		return false;
```
Comment (LingaoM) [inline subsys/bluetooth/mesh/adv_ext.c:329]: Have you considered that `k_work_submit` is not executed immediately, but in an asynchronous manner.

Assuming that there are multiple mesh messages that need to be relayed in an advertising event report(it is possible) then after your change this time, only one ext adv will be scheduled repeatedly. The default method was that multiple relay messages would occupy different ext advs respectively.

**Review by LingaoM — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/mesh/adv_ext.c (hunk 1 above)
Comment (PavelVPV) [reply subsys/bluetooth/mesh/adv_ext.c:329]: Right, I missed that it returns `false` here. Added `k_work_is_pending`.

**Review by LingaoM — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

File: subsys/bluetooth/mesh/adv_ext.c [hunk 2]
```diff
@@ -267,14 +265,8 @@ static void send_pending_adv(struct k_work *work)
 
 			ext_adv->adv = NULL;
 		}
-
-		(void)schedule_send(ext_adv);
```
Comment (alxelax) [inline subsys/bluetooth/mesh/adv_ext.c:271]: Did you run this change over your advertiser suspend test? If I remember correctly, this rescheduling is triggered when you suspend mesh in start callback.

Comment (PavelVPV) [reply subsys/bluetooth/mesh/adv_ext.c:271]: Yes, it passed.

Comment (alxelax) [reply subsys/bluetooth/mesh/adv_ext.c:271]: Ok.

**Review by alxelax — APPROVED** [review]

**Review by Andrewpini — APPROVED** [review]

**Review by akredalen — APPROVED** [review]

**Pull request merged**